src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt

```diff
@@ -181,11 +194,14 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
             if (!checkStateEquivalence) return@also
             val stateEquivalenceCorrect = it.checkStateEquivalenceImplementation()
             if (!stateEquivalenceCorrect) {
-                if (requireStateEquivalenceImplCheck) {
-                    val errorMessage = StringBuilder().appendStateEquivalenceViolationMessage(sequentialSpecification).toString()
-                    error(errorMessage)
-                } else {
-                    reporter.logStateEquivalenceViolation(sequentialSpecification)
+                if (!showedStateEquivalenceError) {
```

Comment:
It's simpler to create a new verifier with `checkStateEquivalence = true` only the first time.